recipes/newrelic/apm/dotNet/debian-systemd-nginx.yml

```diff
@@ -0,0 +1,283 @@
+# Visit our schema definition for additional information on this file format
+# https://github.com/newrelic/open-install-library/blob/main/docs/recipe-spec/recipe-spec.md#schema-definition
+
+name: dotnet-agent-installer
+displayName: .NET Agent Installer
+description: New Relic install recipe for instrumenting .NET applications on Linux
+repository: https://github.com/newrelic/newrelic-dotnet-agent
+
+installTargets:
+  - type: application
+    os: linux
+    platform: "debian"
+  - type: application
+    os: linux
+    platform: "ubuntu"
+
+keywords:
+  - Apm
+  - .NET
+  - dotnet
+  - aspnet
+  - core
+
+processMatch: []
```

Comment:
@gabeobrien @nr-ahemsath @jaffinito 
There is a potential issue here. Without a processMatch this recipe is going to run for every ubuntu/debian linux, which may have nothing to do with .net core.
Is there a process we could latch on to target this recipe when it will be more proper to run?

Reply:
I took a look at a few other recipes and it does seem that the '[]' process match will match in all cases where the host matches.  
I will have to deploy a .Netcore app and see what 'ps' shows to know for sure what value to use for processMatch.

Reply:
We could use `dotnet`, but that only covers cases where customers use `dotnet someawesome.dll` to start the app. If they start it directly from an executable, we won't pick it up.

Reply:
I was thinking the same, and I think like you, think we shouldn't do this (targeting dotnet process).

Reply:
Just want to add a point of clarification for `processMatch: []`

This will _not_ match all cases, and actually does the opposite by matching nothing. The current usage of an empty `processMatch` is limited to the infra-agent and logs recipes, and those are special recipes that have a dedicated execution path through the CLI. The recipes that come after those are surfaced from the `recommendations` endpoint in GraphQL, and that only returns a match when the regex entries in `processMatch` match a given process command the CLI sends to that endpoint ([here](https://github.com/newrelic/newrelic-cli/blob/708c26bfd1fba7887a6fd06e95180fb1a0be00a1/internal/install/recipes/service_recipe_fetcher.go#L74) is where that input is created and a few lines down is where NerdGraph is called).

[Here's the service code](https://source.datanerd.us/open-source/open-install-library-service/blob/master/src/modules/recommendations/recommendations.service.ts#L22-L24) - notice it's designed so that it'll only return a recipe as "recommended" when a process command that the CLI sends matches an entry in `processMatch`. 

For a concrete example, view [GraphiQL query with processDetails](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20docs%20%7B%0A%20%20%20%20openInstallation%20%7B%0A%20%20%20%20%20%20recommendations%28criteria%3A%20%7BprocessDetails%3A%20%5B%7Bname%3A%20%22java%22%7D%5D%2C%20installTarget%3A%20%7Btype%3A%20APPLICATION%2C%20os%3A%20LINUX%2C%20platform%3A%20DEBIAN%7D%7D%29%20%7B%0A%20%20%20%20%20%20%20%20count%0A%20%20%20%20%20%20%20%20results%20%7B%0A%20%20%20%20%20%20%20%20%20%20installTargets%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20os%0A%20%20%20%20%20%20%20%20%20%20%20%20type%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A) vs [GraphQL query without processDetails](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20docs%20%7B%0A%20%20%20%20openInstallation%20%7B%0A%20%20%20%20%20%20recommendations%28criteria%3A%20%7BprocessDetails%3A%20%5B%5D%2C%20installTarget%3A%20%7Btype%3A%20APPLICATION%2C%20os%3A%20LINUX%2C%20platform%3A%20DEBIAN%7D%7D%29%20%7B%0A%20%20%20%20%20%20%20%20count%0A%20%20%20%20%20%20%20%20results%20%7B%0A%20%20%20%20%20%20%20%20%20%20installTargets%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20os%0A%20%20%20%20%20%20%20%20%20%20%20%20type%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A).

Not sure how helpful that is to the matter at hand, besides confirming that you _will_ need to include something in `processMatch` for this to work.

Reply:
@jbeveland27 I guess I got the processMatch backward, but either way, something needs to be enhanced. We want the recipe to be picked up for the correct folks.
Thanks for clarifying.